pinot-core/src/main/java/org/apache/pinot/core/transport/AsyncQueryResponse.java

```diff
@@ -39,6 +39,8 @@
   private final CountDownLatch _countDownLatch;
   private final long _maxEndTimeMs;
 
+  private Exception _brokerRequestSendException;
```

Comment:
Make it volatile? (Although it is always set and read by the same thread)

Reply:
Good suggestion. Being on same thread might change in future due to some refactoring and it's good to have it as volatile.